### ISSUE TITLE
feat(aws-cdk-lib/aws-kms): Allow testing for the existence of a KMS key by alias name

### ISSUE THREAD

**Issue by neoakris** [aws/aws-cdk#31574]
### Describe the feature

**Here's a code snippet for some context:**
```typescript
import * as kms from 'aws-cdk-lib/aws-kms';
let kmsKey = kms.Key.fromLookup(stack, "pre-existing-kms-key", { aliasName: "alias/eks/test" })
```

**Current State of CDK 2.133.0:**  
* The above logic works when a kms-key with alias "eks/test" exists.
* If no key exists, then the program stops with error  
  `[Error at /$STACK_NAME] Could not find any key with alias named eks/test`  
    I tried wrapping the above in try catch logic, but the program still does a hard stop with the above error as soon as it discovers the key doesn't exist. 

**Feature Request/Proposed Solution:**
* Update logic:  
   If no key exists, instead of an error, return undefined. This will make it easier to allow if statements to handle that scenario. 

### Use Case

While using EKS Blueprints, every time I delete and recreate a cluster (semi-frequently for ephemeral sandbox / test environments), due to its defaults it'll create a new KMS key each time, then on delete I get orphaned kms keys. 
EKS Blueprints allows passing in a pre-existing kms key to avoid this. 

However I'd like to add logic for the following use case:
1. specify a desired kms alias
2. detect if it exists
3. if not exist: create new key with kms alias, and use the newly created kms key.
4. if exists: use kms key with alias.

Basically I wanted to be able to create a function named ensure_existence_of_kms_key_with_alias()

This will allow me to implement logic where when I create a cluster it'll create a kms key, but if I delete and recreate the cluster it'll reuse the original kms key. (This avoids orphaned keys, and is also advantageous in the scenario of if disk backups were ever created using a kms key, then restore might be easier if a consistent kms key were used.)

### Other Information

The proposed feature/solution shouldn't be a breaking change.

### Acknowledgements
- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.133.0 (build dcc1e75)

### Environment details (OS name and version, etc.)

MacOS Sonoma 14.6.1

**Comment by khushail** [comment]
Hi @neoakris , thanks for reaching out. I think this can be acheived using a flag type variable, say, KeyAliasExists(), which would return `false` if its not. Your usecase makes sense to me.

However if you would like to contribute, please feel free to follow our [contribution guide](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-1-find-something-to-work-on). I would be marking this feature request as P3 for keeping it open for community contribution as well. Thanks.

**Comment by neoakris** [comment]
You're right a flag makes more sense, it'd be just as effective as a string comparison, yet cleaner.

Thanks for the triage and suggestions.

**Comment by go-to-k** [comment]
Hi @khushail @neoakris ,

I have submitted the [PR](https://github.com/aws/aws-cdk/pull/31676), where the `fromLookup` method returns a dummy key for now.

The reasons for this are:

- Originally, the method used the [`dummyValue`](https://github.com/aws/aws-cdk/blob/v2.161.0/packages/aws-cdk-lib/aws-kms/lib/key.ts#L685-L687) for [**when the key was not found**](https://github.com/aws/aws-cdk/blob/v2.161.0/packages/aws-cdk-lib/core/lib/context-provider.ts#L32-L36) when getting the key from the context provider, but the value was not used in the destination process (that's context provider) and an error had been occurred.
- Just recently, [a feature was added to the SSM parameter context method](https://github.com/aws/aws-cdk/pull/31415) that allows it to ‘return a dummy value if the parameter does not exist’. This is very similar to the requirement in this case. There, the new option [`IgnoreErrorOnMissingContext`](https://github.com/aws/aws-cdk/blob/v2.161.0/packages/aws-cdk-lib/aws-ssm/lib/parameter.ts#L587) has been added to the ContextProvider options and is used.

Allowing `undefined` in the return value of `fromLookup` is a breaking change, so I decided not to do that. I also considered allowing a dummy value to be freely specified (like the SSM Parameter case), but decided against that too as I didn't think there would be many use cases for it.

Of course, a new method can be created that is separate from the `fromLookup` method, such as `Key.keyAliasExists()`.
However, given the design philosophy of the original code, I thought it would be better to fix it to the correct implementation of `fromLookup`.

Please let me know if you have any feedback.

**Comment by neoakris** [comment]
@go-to-k my understanding of what you've said is once implemented/in a live release, if I specify the returnDummyKeyOnMissing option. I'll be able to do an if statement check against 1234abcd-12ab-34cd-56ef-1234567890ab when the key isn't pre-existing that works for me. 

Thanks for the update!

**Comment by go-to-k** [comment]
That's right.

**Comment by khushail** [comment]
@go-to-k , your implementation with the existing design is very thoughful. Thanks for your efforts in submitting a PR!

**Comment by go-to-k** [comment]
@khushail Thank you! The `pr/needs-cli-test-run` label has been attached, what should I do? If I just wait, will the maintainer run the test for me?

**Comment by khushail** [comment]
@go-to-k , I am not really sure why this label is attached. You could ask in the PR from the maintainer.
